prijateli_tree/app/routers/utils/games_utils.py

```diff
@@ -185,3 +185,87 @@ def get_game_and_type(game_id: int, db: Session = Depends(get_db)):
     raise_exception_if_none(game, detail="game not found")
 
     return game, game.game_type
+
+
+def get_header_data(player: GamePlayer, db: Session = Depends(get_db)):
+    """
+    Gets the player's score, name and game progress
+    """
+    score_dict = get_score_and_name(player, db)
+    progress_dict = get_games_progress(player, db)
+
+    return {**score_dict, **progress_dict}
+
+
+def get_score_and_name(player: GamePlayer, db: Session = Depends(get_db)):
+    """
+    Gets the player's score and name from the session player object
+    by using the game player object
+    """
+    session_player = get_session_player_from_player(player, db)
+    player_name = f"{player.user.first_name} {player.user.last_name}"
+    player_score = session_player.points
+
+    return {"player_name": player_name, "player_score": player_score}
+
+
+def get_games_progress(player: GamePlayer, db: Session = Depends(get_db)):
+    """
+    Gets the player's progress in the overall session
+    """
+    session_player = get_session_player_from_player(player, db)
+    session_id = session_player.session_id
+
+    # Get game ids
+    practice_game_ids = (
+        db.query(Game)
+        .filter_by(game_session_id=session_id, practice=True)
+        .order_by(Game.id)
+        .all()
+    )
+    num_practice_games = len(practice_game_ids)
+
+    real_game_ids = (
+        db.query(Game)
+        .filter_by(game_session_id=session_id, practice=False)
+        .order_by(Game.id)
+        .all()
+    )
+    num_real_games = len(real_game_ids)
+
+    # Select completed games by player
+    completed_games = (
+        db.query(GamePlayer)
+        .filter_by(session_player_id=session_player.id, completed_game=True)
+        .all()
+    )
+
+    # Get number of completed games
+    completed_practice_games = 0
+    completed_real_games = 0
+
+    for real_game in real_game_ids:
+        if real_game in completed_games:
+            completed_real_games += 1
+
+    for practice_game in practice_game_ids:
```

Comment:
It will be easier to figure out the current round using the length of lists and the logic of the game -- ie. the practice rounds come first, then the real games.  session_players are unique to a given session, so I don't think we need to worry about the list lengths referring to games that don't fit our logic.

Reply:
Play through a few rounds of the game and make sure the numbers update the way you think they will.